packages/extension-vscode/src/server.ts

```diff
@@ -0,0 +1,204 @@
+    ProposedFeatures
+} from 'vscode-languageserver';
+
+// TODO: Enhance `hint` exports so everything can be imported directly.
+import * as hint from 'hint';
+import * as config from 'hint/dist/src/lib/config';
+import * as loader from 'hint/dist/src/lib/utils/resource-loader'; // eslint-disable-line
+import { Problem, Severity, UserConfig } from 'hint/dist/src/lib/types';
+
+// Connect to the language client
+const connection = createConnection(ProposedFeatures.all);
+const documents = new TextDocuments();
+
+/* istanbul ignore next */
+const trace = (message: string): void => {
+    return console.log(message);
+};
+
+const loadModule = async <T>(context: string, name: string): Promise<T> => {
+    return await Files.resolveModule2(context, name, null, trace);
+};
+
+// Load a user configuration, falling back to 'development' if none exists.
+const loadUserConfig = (directory: string, Configuration: typeof config.Configuration): UserConfig => {
+    const defaultConfig: UserConfig = { extends: ['development'] };
+
+    try {
+        const configPath = Configuration.getFilenameForDirectory(directory);
+        const resolvedPath = path.resolve(directory, configPath);
+
+        return Configuration.loadConfigFile(resolvedPath) || defaultConfig;
+    } catch (e) {
+        return defaultConfig;
+    }
+};
+
+// Load a copy of webhint with the provided configuration
+const loadEngine = async (directory: string, configuration: config.Configuration): Promise<hint.Engine> => {
+    const { loadResources } = await loadModule<typeof loader>(directory, 'hint/dist/src/lib/utils/resource-loader');
+    const resources = loadResources(configuration);
+    const { Engine } = await loadModule<typeof hint>(directory, 'hint');
+
+    return new Engine(configuration, resources);
+};
+
+// Load both webhint and a configuration, adjusting it as needed for this extension
+const loadWebHint = async (directory: string): Promise<hint.Engine> => {
+    const { Configuration } = await loadModule<typeof config>(directory, 'hint/dist/src/lib/config');
+    const userConfig = loadUserConfig(directory, Configuration);
+
+    // The vscode extension only works with the local connector
+    userConfig.connector = { name: 'local' };
+
+    if (!userConfig.parsers) {
+        userConfig.parsers = [];
+    }
+
+    // Ensure the HTML parser is loaded
+    if (userConfig.parsers.indexOf('html') === -1) {
+        userConfig.parsers.push('html');
+    }
+
+    return await loadEngine(directory, Configuration.fromConfig(userConfig));
+};
+
+let workspace: string;
+let engine: hint.Engine;
+
+connection.onInitialize(async (params) => {
+    try {
+        // TODO: support multiple workspaces (`params.workspaceFolders`)
+        workspace = params.rootPath;
+        engine = await loadWebHint(workspace);
+    } catch (e) {
+        console.error(e);
+        connection.window.showErrorMessage('[webhint] Load failed. Add it via `npm install hint --save-dev`.');
```

Comment:
What if the user has webhint installed but not the default configuration? Will we be printing the same error message? Is there a way we can detect different errors?
Also where is `console.error(e)` going to output?

Reply:
Probably, but I can look into differentiating these. And `console.error(e)` is just there for debugging - it shows up in VS Code's "Output" panel for the extension (Go to "View > Output", then pick the extension name from the drop down in the panel).